### ISSUE TITLE
TXXX AuxData extraction from MP3 showing subtag in value

### ISSUE THREAD

**Issue by imolb** [gerbera/gerbera#1410]
With #421 the user defined text frame TXXX was integrated.

I've two issues with this functions:
1. The name of the subtag is included in the value.
2. Multiple separated values are concatenated.

To 1) With MusicBrainz I've added the tag "TXXX:MusicBrainz Album Type" with the value "album". Gebera extracts the value "MusicBrainz Album Type album" for this tag. I would have expected only "album".
To 2) MusicBrainz allows to define several values and separates these with 0x00 00 FF FE. Gerbera seems to pack the values together and separates these with a space. (Maybe the multiple values might be outside of the mp3 specification.)

[sample_releasetype.zip](https://github.com/gerbera/gerbera/files/6332001/sample_releasetype.zip)

sample_releasetype_one.mp3:
`debug: [taglib_handler.cc:478] extractMP3(): Adding auxdata: TXXX:MusicBrainz Album Type with value MusicBrainz Album Type album`

sample_releasetype_multi.mp3
`debug: [taglib_handler.cc:478] extractMP3(): Adding auxdata: TXXX:MusicBrainz Album Type with value MusicBrainz Album Type album compilation`

Tested with recent v1.8.0-33-g48db6739.

**Comment by KarlStraussberger** [comment]
I just added your files to my test rep. I cannont reproduce your value "MusicBrainz Album Type album". Where did you check the value? Here's the content of the details in web UI.
![grafik](https://user-images.githubusercontent.com/23654788/115197497-55ae9780-a0f1-11eb-85ab-3b86f7153a0a.png)

Multivalues are a complicated thing because there is no real standard how to separate them. You may try with `<library-options multi-value-separator="; " legacy-value-separator="???">` if you are able to encode the MusicBrainz separator (???) in UTF-8. Anyway there is no such thing as multi-value-storage in gerbera at the moment.

**Comment by imolb** [comment]
Thanks for the fast reply!

My details in the web UI looks different, especially in the important attribute:

![image](https://user-images.githubusercontent.com/49784529/115255761-df308a80-a12e-11eb-87a3-c112e64b91a7.png)

I also enabled the debug mode and uncommented some log messages in tablib_handler.cc (line 471 was additionally inserted by me).
```
2021-04-19 14:38:52.142  debug: [taglib_handler.cc:455] extractMP3(): Adding auxdata: TIT2 with value sample
2021-04-19 14:38:52.142  debug: [taglib_handler.cc:459] extractMP3(): TXXX Frame list has 1 elements
2021-04-19 14:38:52.142  debug: [taglib_handler.cc:471] extractMP3(): TXXX Raw value: [MusicBrainz Album Type] MusicBrainz Album Type album
2021-04-19 14:38:52.142  debug: [taglib_handler.cc:475] extractMP3(): TXXX Tag: MusicBrainz Album Type
2021-04-19 14:38:52.142  debug: [taglib_handler.cc:478] extractMP3(): Adding auxdata: TXXX:MusicBrainz Album Type with value MusicBrainz Album Type album
2021-04-19 14:38:52.142  debug: [taglib_handler.cc:259] fillMetadata(): TagLib handler done.

```
Somehow looks like the subtag name "MusicBrainz Album Type" is reported redundant in the value.

I'm using taglib-1.11.1-r3.

**Comment by KarlStraussberger** [comment]
It depends on taglib version. Your tag is know in taglib 1.12, so it is stripped. We can do the same:

```
                if (desiredSubTag == subTag) {
                    if (content.substr(0, subTag.length()) == subTag)
                        content = content.substr(subTag.length());

                    log_debug("Adding auxdata: {} with value {}", desiredFrame.c_str(), content.c_str());
                    item->setAuxData(desiredFrame, content);
                    break;
                }
```

I don't really like it, but  it work. I'm not sure whether there are situations out there that break an import.

**Comment by imolb** [comment]
I'm a little bit confused. Do you mean I need taglib 1.12 _or_ your fix? Or both? Or can I use with taglib 1.12 directly the tag "RELEASETYPE" in the config.xml and import.js without your fix?

But at least I can confirm:
Your fix together with taglib 1.12 works for me with tag "TXXX:MusicBrainz Album Type".
With taglib 1.11.1-r3 it works partly, but the content has a leading space.

**Comment by KarlStraussberger** [comment]
Taglib 1.12 does not need the fix. But I think we should add the fix (without leading space) for other tags and folks still on 1.11.1.
